app/device/gps.py

```diff
@@ -75,24 +80,29 @@ def set_datatype(self, datatype):
 
         self.ser.write(DATATYPE_MAPPINGS[datatype])
 
-    @classmethod
+
+    @staticmethod
     def parse_raw_data(raw_data):
```

Comment:
saying this means we call the method on the class instead of the object (ie. `GPS.parse_raw_data`)...this makes testing you logic easier

Reply:
@NikhilPeri, do you mean to say call the method on the object instead of the class? Or am I understanding something incorrectly?

I thought the GPS class would implement the singleton pattern so that only one object of it can be made - and then all methods would call on the sole object of the GPS class. I feel like I'm missing something; feel free to correct my thinking.

Reply:
This is seperate from the Singleton Pattern, although both will be static class methods.
 - The `parse_raw_data` method is a static class method so it can be called without an instance of the object.  This is ideal for testing since we can test the functionality of the parse without having to pull data from a sensor (ie. i can test on my laptop instead of the pi)
 - The singletone pattern (not implemented yet) will also be a static class method so we can call `getDevice` on the class and get an object..it will be implemented in a way that calling ti multiple times will always return a reference to the same object